src/components/Modal/Modal.js

```diff
@@ -0,0 +1,64 @@
+import React, { useEffect, useRef } from 'react';
+import cx from 'classnames';
+
+import './Modal.scss';
+
+const CN = 'modal';
+
+const Modal = (props) => {
+  const {
+    show,
+    children,
+    hideModal,
+    className
+  } = props;
+  const modalRef = useRef(null);
+
+  useEffect(() => {
+    if (show) {
+      document.body.classList.add('remove-scroll');
+      modalRef.current.focus();
+    }
+  });
```

Comment:
Do you think, that you are prepered to use hooks?

Reply:
it is valid usage of use effect, but from performance perspective it could be a little better if you pass show as deps variable

```useEffect(() => {
      document.body.classList. toggle('remove-scroll');
      modalRef.current.focus();
  }, [show]);```

but this is not critical

the only difference is, that without deps argument (second one) use effect will run on each rerender of the component, passing show as arg prevents unnecessary checks

Reply:
I agree! But in limitations of this project @KhrystynaKovtun you can stick with class-component implementation